goty-client/src/app/form/giveaway/giveaway.component.ts

```diff
@@ -1,19 +1,24 @@
-import {Component} from '@angular/core'
+import {Component, Input} from '@angular/core'
 import {lastTime} from 'src/api/constants'
 import {AppService} from '../../../api/app/app.service'
-import {AppQuery} from '../../../api/app/app.store'
 
 @Component({
   selector: 'app-giveaway',
   templateUrl: './giveaway.component.html',
   styleUrls: ['./giveaway.component.scss']
 })
 export class GiveawayComponent {
+  @Input() enteredGiveaway
+  @Input() readonly = false
   title = 'Giveaway'
   lastTime = lastTime
-  enteredGiveaway$ = this.appQuery.selectEnteredGiveaway()
 
-  constructor(public readonly appService: AppService,
-              private readonly appQuery: AppQuery) {
+  constructor(public readonly appService: AppService) {
+  }
+
+  setEnteredGiveaway(enteredGiveaway: boolean) {
+    if (!this.readonly) {
```

Comment:
Consider abstracting the logic to the parent component.

Reply:
done